src/v/kafka/server/handlers/list_offsets.cc

```diff
@@ -318,16 +320,21 @@ list_offsets_handler::handle(request_context ctx, ss::smp_service_group ssg) {
       std::make_move_iterator(unauthorized_it),
       std::make_move_iterator(request.data.topics.end()));
 
-    request.data.topics.erase(unauthorized_it, request.data.topics.end());
+    request.data.topics.pop_back_n(
+      std::distance(unauthorized_it, request.data.topics.end()));
 
     list_offsets_ctx octx(
       std::move(ctx), std::move(request), ssg, std::move(unauthorized_topics));
 
     return ss::do_with(std::move(octx), [](list_offsets_ctx& octx) {
         auto topics = list_offsets_topics(octx);
-        return when_all_succeed(topics.begin(), topics.end())
+        return ss::when_all_succeed(topics.begin(), topics.end())
           .then([&octx](std::vector<list_offset_topic_response> topics) {
-              octx.response.data.topics = std::move(topics);
+              octx.response.data.topics.reserve(topics.size());
+              std::move(
+                topics.begin(),
+                topics.end(),
+                std::back_inserter(octx.response.data.topics));
```

Comment:
I think it would be clearer like this, since the `unauthorized_topics` is roughly the same type:
```suggestion
              octx.response.data.topics = {
                std::make_move_iterator(topics.begin()),
                std::make_move_iterator(topics.end())};

Reply:
yeah good point